src/modules/LinearTracking.jl

```diff
@@ -125,32 +125,151 @@ function linear_solenoid_matrix(Ks, L)
 end
 
 
-function linear_bend_matrices(K0, L, gamma_0, e1=nothing, e2=nothing)
-  theta = K0*L
-  s, c = sincos(theta)
-  cc = (sincu(theta/2)^2)/2
-  sc = sincu(theta)
-  mx = SA[c  L*sc; -K0*s  c]
-  my = SA[1  L; 0 1]
-  r56 = L*(1/gamma_0^2 - theta^2*sincuc(theta))
-  d = SA[theta*L*cc, theta*sc, 0, 0]
-  t = SA[-theta*sc,  -theta*L*cc, 0, 0]
-
-  if !isnothing(e1) && e1 != 0
-    me1 = K0*tan(e1)
-    mx = mx*SA[1 0; me1  1]
-    my = my*SA[1 0; -me1 1]
-    t = SA[t[1]+me1*t[2], t[2], 0, 0]
-  end
 
-  if !isnothing(e2) && e2 != 0
-    me2 = K0*tan(e2)
-    mx = SA[1 0; me2  1]*mx
-    my = SA[1 0; -me2 1]*my
-    d = SA[d[1], me2*d[1]+d[2], 0, 0]
-  end
+function linear_dipole_matrices(K0, L, gamma_0; g::Union{Nothing, Real}=nothing, K1::Union{Nothing, Real}=nothing, e1=nothing, e2=nothing)
```

Comment:
```suggestion
function linear_dipole_matrices(g, e1, e2, K0, K1, gamma_0, L)
```
Can we just use this for this function definition instead? I think that will simplify things a lot. Then the code will just check if they are zero and do the corresponding operations.

The idea of using `nothing` was to get some compile-time optimizations, but it is becoming too complicated and with any type instability we can get big performance hits

Reply:
I think this approach works as long as we ensure we don’t assign `nothing` to these values.

Would it perhaps be better to default `K0` and `K1` to `0` instead of `nothing`? That way, we wouldn’t need to specify them every time when they’re not used. Since `e1` and `e2` don’t have type instability issues, I think it makes sense to keep them as they are.

Reply:
Let's let all of them default to `0`, and keep the checks `if __ \approx 0`